news/_posts/2019-09-11-prestashop-in-2019-and-beyond-part-2-pain-points.md

```diff
@@ -0,0 +1,475 @@
+---
+layout: post
+title:  "PrestaShop in 2019 and beyond, part 2: The Pain Points"
+subtitle: "aka What needs to be improved"
+date:   2019-09-11 09:30:00
+authors: [ PabloBorowicz ]
+icon: icon-compass
+tags: [1.7, architecture]
+---
+
+This is the second in a [series of articles][introduction] we introduced earlier this year, that aims to describe where we are, where we are going, and some ideas on how we'll get there.
+
+# The Pain points
+_(or "What needs to be improved")_
+
+![The only valid measurement of code quality: WTFs/minute](/assets/images/2019/09/wtf-per-minute.png)
+{: .text-center style="margin: 5rem auto"}
+
+In the [previous part][part-1], we described what the current architecture looks like. This article aims to analyze what are the main "pain points", that is, fundamental problems that are dragging the project down.
+
+But before getting to that, let's address the elephant in the room. After reading the previous part in this series, probably the number one thing you thought of was "why _the hell_ is this so complicated?". 
+
+There are many reasons for that, but I think there's one that can be found at the root of it. This will be a long article, so please bear with me.
+
+## Introduction
+
+First of all, I believe there is an all-too-common misunderstanding about the very nature of PrestaShop. Many people think of PrestaShop as a product: something that you download and install to build and run your shop from your browser. But it's not. Or at least I don't see it that way.
+
+When I meet people and tell them what I do for a living, I usually describe PrestaShop as an _open-source platform for developing e-commerce websites_. This is in my view, a more accurate description of what this project is about.
+
+Yes, I have met people who download it and use it as-is, but I think they belong to a declining minority. To my best understanding, most people who are choosing to use the PrestaShop project in 2019 (and who succeed in doing so) do it to kickstart their own development. They are usually agencies or experts, and often they are also part of our rich community of module and theme developers who sell on the [Addons Marketplace][addons].
```

Comment:
Wdyt @ttoine ?

```suggestion
Yes, I have met people who download it and use it as-is, but I think they belong to a declining minority. To my best understanding, most people who are choosing to use the PrestaShop project in 2019  and succeed in doing so, do it to kickstart their own development—that is, a _product_ based on the PrestaShop _project_ (remember [not to confuse projects and products](https://opensource.com/business/15/8/open-source-products-four-rules)). They are usually agencies or experts, and often they are also part of our rich community of module and theme developers who sell on the [Addons Marketplace][addons].
```

Reply:
I think that you can add something like "Actually, even someone who is downloading PrestaShop to start his own shop, will create his own custom product, with combination of modules, a theme, and configuration. Each and every PrestaShop store is unique and is a product based on the project".

(or maybe it's a bit too much?)

Reply:
I agree with what you're saying, but I think that would be straying too far away from my main argument (that there are several stakeholders and developers are my main concern).

I don't know how to further develop the distinction between project and product without confusing people regarding my main idea.

Reply:
ok
let's use your proposal as it is. and let see what kind of feedback we get (will there be some confusion?)

Reply:
Okay! We'll rephrase if needed.